appcues/src/main/java/com/appcues/AppcuesFirebaseMessagingService.kt

```diff
@@ -0,0 +1,88 @@
+package com.appcues
+
+import android.Manifest.permission
+import android.content.Context
+import android.content.pm.PackageManager
+import androidx.core.app.ActivityCompat
+import androidx.core.app.NotificationCompat
+import com.appcues.util.notify
+import com.appcues.util.setContentIntent
+import com.appcues.util.setStyle
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+/**
+ * Appcues Firebase Messaging Service implementation.
+ *
+ * To use it you can either add directly to your Manifest as a service or
+ * call handleMessage and setToken on your custom implementation.
+ */
+public class AppcuesFirebaseMessagingService : FirebaseMessagingService() {
+
+    public companion object {
+
+        /**
+         * handleMessage will try to parse the received message into an Appcues notification, if it does it will return true,
+         * or false in case the message is not for Appcues to handle
+         *
+         * @param context application context ('this' when calling from service)
+         * @param message remote message coming from the messaging service
+         *
+         * @return determining whether we handled the message or not.
+         */
+        @JvmStatic
+        public fun handleMessage(context: Context, message: RemoteMessage): Boolean {
+            // check for permission
+            if (ActivityCompat.checkSelfPermission(context, permission.POST_NOTIFICATIONS) != PackageManager.PERMISSION_GRANTED)
+                return false
+
+            val data = try {
+                AppcuesMessagingData(message)
+            } catch (_: IllegalStateException) {
+                return false
+            }
+
+            NotificationCompat.Builder(context, "channel")
+                .setContentTitle(data.title)
+                .setContentText(data.body)
+                .setAutoCancel(true)
+                .setStyle(data)
+                .setContentIntent(context, data)
+                .notify(context)
+
+            return true
+        }
+
+        /**
+         * sets the global appcues token
+         *
+         * @param token token value coming from the messaging service
+         */
+        @JvmStatic
+        public fun setToken(token: String) {
+            Appcues.pushToken = token
+        }
+    }
+
+    internal class AppcuesMessagingData(message: RemoteMessage) {
+
+        // required
+        val title: String = message.data["title"] ?: throw IllegalStateException("title not found")
+        val body: String = message.data["body"] ?: throw IllegalStateException("body not found")
+        val notificationId: String =
+            message.data["appcues_attachment_url"] ?: throw IllegalStateException("appcues_attachment_url not found")
```

Comment:
I believe this should be looking for `appcues_notification_id` rather than attachment